chia/wallet/wallet_state_manager.py

```diff
@@ -1493,9 +1488,20 @@ async def _add_coin_states(
                                     for rem_coin in tx_record.removals:
                                         if rem_coin == coin_state.coin:
                                             confirmed_tx_records.append(tx_record)
-
                             for tx_record in confirmed_tx_records:
                                 await self.tx_store.set_confirmed(tx_record.name, uint32(coin_state.spent_height))
+                            if record.coin_type == CoinType.CLAWBACK:
+                                await self.interested_store.remove_interested_coin_id(coin_state.coin.name())
+                                if len(confirmed_tx_records) == 1 and record.metadata is not None:
```

Comment:
can you explain what's happening here

Reply:
This is the weird issue I mentioned in the engineering channel. After the user sends clawback to another wallet and claims/clawback then starts the resync, the wallet will receive 1 coin creation event and 1 coin spent event. Usually, during the resync, we will only receive 1 coin spent event. This weird behavior will cause we cannot recover the Clawback_Outgoing tx.
To fix this issue, I added this part. For a spent clawback coin which we already saved it in the DB, we expect there are two TXs need to be updated. One for CLAWBACK_INCOMING and another for CLAWBACK_OUTGOING. If we only find 1 TX then it means we don't have the outgoing TX. So in this case, we want to add a CLAWBACK_OUTGOING TX.
Actually, this is a general issue for other assets too. It means the wallet is not able to recovery 100% of outgoing TXs.

Reply:
can you please add this comment as a comment in the code

Reply:
I spent few hours now looking into this and it feels now more like hack than a fix. Since the trusted test passes without this block we should imo rather better understand the issue why the untrusted doesn't behave the same than just adding this block here. What i know so far is the difference is that we don't long sync in this case for untrusted but instead end up doing a short sync via `sync_from_untrusted_close_to_peak` and we even end up with this edge case here https://github.com/Chia-Network/chia-blockchain/blob/88a692666d737f56a74b8241a36e89d508bd3343/chia/wallet/wallet_node.py#L1174-L1190 because the chain isn't long enough for a weight proof.